src/query/functions/temporal/base.go

```diff
@@ -348,53 +415,65 @@ func (c *baseNode) processSingleRequest(
 	}
 
 	aggDuration := c.op.duration
-	steps := int((aggDuration + bounds.Duration) / bounds.StepSize)
-	values := make([]ts.Datapoints, 0, steps)
-	desiredLength := int(math.Ceil(float64(aggDuration) / float64(bounds.StepSize)))
+	depIters := make([]block.UnconsolidatedSeriesIter, len(request.deps))
```

Comment:
For safety, should we just use `make([]foo, 0, length)` and `append(...)` wherever it's possible to?  Been running into a lot of these bugs lately (in not just M3 code base, elsewhere too).